svc.go

```diff
@@ -133,6 +166,10 @@ func (as *AutoNATService) handleDial(p peer.ID, obsaddr ma.Multiaddr, mpi *pb.Me
 			continue
 		}
 
+		if ip, err := addrToIP(addr); err != nil || !bytes.Equal(obsHost, ip) {
```

Comment:
https://golang.org/pkg/net/#IP.Equal

Reply:
Is this change to ensure that the IP address we dial the AutoNAT client back on is the same as the address with which it has connected to us ?

If we do not do this, what kind of a problem do we run into ?

Reply:
If we don't do this, a client could ask an autoNat service to dial arbitrary host/port combinations.